app/common/src/services/Backend.ts

```diff
@@ -995,6 +1004,11 @@ export function isPlaceholderId(id: AssetId) {
   return typeof id !== 'string' && PLACEHOLDER_SIGNATURE in id
 }
 
+/** Whether a given asset represents a credential. */
+export function isAssetCredential(asset: Asset) {
+  return asset.serviceName !== undefined
+}
```

Comment:
Currently used to disallow editing a credential as plaintext secret (which is invalid). In the future we will have some refresh logic for credentials, but for now we just disallow editing them altogether.